### PR TITLE
Win service needs to set ContentRootPath

### LINKED ISSUE #26843: Please review content for compatibility with .net 6
I think some of this content need to be reviewed for compatibility with .net 6.  UseWindowsService() was throwing a not supported exception for me.  A couple other posts helped me work around the issue.

https://stackoverflow.com/questions/72932609/net-6-usewindowsservice-throws-exception
https://stackoverflow.com/questions/70571849/host-asp-net-6-in-a-windows-service

[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b0bd8cb8-7e28-8c4d-a550-3aea8f3ab4dd
* Version Independent ID: 4b60d24e-ef05-15e5-10b4-483094081bf3
* Content: [Host ASP.NET Core in a Windows Service](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/windows-service?view=aspnetcore-6.0&tabs=visual-studio#app-configuration)
* Content Source: [aspnetcore/host-and-deploy/windows-service.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/windows-service.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #26843
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/windows-service?view=aspnetcore-7.0&branch=pr-en-us-26844&tabs=visual-studio#app-configuration)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@tothalexlaszlo please review

**Review by tothalexlaszlo — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@halter73  44 second review please.

File: aspnetcore/host-and-deploy/windows-service/samples/6.x/WebAppServiceSample/Program.cs [hunk 1]
```diff
@@ -4,7 +4,8 @@
 var options = new WebApplicationOptions
 {
     Args = args,
-    ContentRootPath = WindowsServiceHelpers.IsWindowsService() ? AppContext.BaseDirectory : default
+    ContentRootPath = WindowsServiceHelpers.IsWindowsService()
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/windows-service/samples/6.x/WebAppServiceSample/Program.cs:7]: This is the snippet referenced in the preceding file.

**Review by halter73 — APPROVED** [review]
This looks good to me. In .NET 7, you will no longer have to manually change `WebApplicationOptions.ContentRootPath` in windows services because of https://github.com/dotnet/runtime/pull/68580.

That PR also adds `builder.Services.AddWindowsService()` which does the same thing as `builder.Host.UseWindowsService()` without using the `builder.Host`, `IHostBuilder` compatibility layer but with less indirection. `AddWindowsService()` also works with the `HostApplicationBuilder` we added in .NET 7.

**Pull request merged**